### PR TITLE
feat(json-rpc): add new method isTransactionIndexedOnNode

### LINKED ISSUE #8526: feat(json-rpc): query existence of transaction on the node from the json-rpc
## Description

Implement `iota_isTransactionIndexedOnNode`.

### PR DESCRIPTION
**Pull request by kodemartin**
# Description of change

This patch adds a new endpoint `iota_isTransactionIndexedOnNode`

## Links to any relevant issues

Fixes #8526

## How the change has been tested

Describe the tests that you ran to verify your changes.

Make sure to provide instructions for the maintainer as well as any relevant configurations.

- [x] Basic tests (linting, compilation, formatting, unit/integration tests)
- [x] Patch-specific tests (correctness, functionality coverage)

### Infrastructure QA (only required for crates that are maintained by @iotaledger/infrastructure)

- [ ] Synchronization of the indexer from genesis for a network including migration objects.
- [ ] Restart of indexer synchronization locally without resetting the database.
- [ ] Restart of indexer synchronization on a production-like database.
- [ ] Deployment of services using Docker.
- [x] Verification of API backward compatibility.

### Release Notes

- [ ] Protocol:
- [ ] Nodes (Validators and Full nodes):
- [ ] Indexer:
- [x] JSON-RPC: A new method `iota_isTransactionIndexedOnNode` is added that allows clients and dapps to optionally poll the fullnode to verify that a transaction has been indexed. The latter is a necessary condition so that any dependent transaction is executed successfully. The new method complements the recent advancements in the Indexer JSON-RPC API that  indexes transactions after execution without waiting for a checkpoint to include them. 
- [ ] GraphQL:
- [ ] CLI:
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

File: crates/iota-json-rpc/src/read_api.rs [hunk 1]
```diff
@@ -727,6 +727,12 @@ impl ReadApiServer for ReadApi {
         .await
     }
 
+    #[instrument(skip(self))]
+    async fn is_transaction_indexed_on_node(&self, digest: TransactionDigest) -> RpcResult<bool> {
+        let transaction = self.get_transaction_block(digest, None).await?;
+        Ok(transaction.digest == digest)
```
Comment (tomxey) [inline crates/iota-json-rpc/src/read_api.rs]: The behaviour seems a bit strange to me. The function name suggests that it returns if transactions is indexed on the node, so true if it is, and false if it's not.
In other words: if user asks a question IF transaction is indexed, he should expect yes or no as normal answers, so I think there should be no error returned in such standard cases.

Have you considered implementing it like that, or was it problematic for some other reasons?

File: crates/iota-json-rpc-tests/tests/read_api.rs [hunk 2]
```diff
@@ -550,6 +550,27 @@ async fn get_transaction_block() {
     get_transaction_block_with_options(IotaTransactionBlockResponseOptions::default()).await;
 }
 
+#[sim_test]
+async fn is_transaction_present() {
```
Comment (tomxey) [inline crates/iota-json-rpc-tests/tests/read_api.rs:570]: could we also test negative case with tx not indexed?

File: crates/iota-json-rpc/src/read_api.rs (hunk 1 above)
Comment (kodemartin) [reply crates/iota-json-rpc/src/read_api.rs]: That was an oversight. Good catch! Fixed with 111203a2be

File: crates/iota-json-rpc-tests/tests/read_api.rs (hunk 2 above)
Comment (kodemartin) [reply crates/iota-json-rpc-tests/tests/read_api.rs:570]: 111203a2be

**Review by Thoralf-M — APPROVED** [review]

**Review by tomxey — APPROVED** [review]
